packages/world-modules/src/modules/puppet/PuppetMaster.sol

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { SystemRegistry } from "@latticexyz/world/src/codegen/tables/SystemRegistry.sol";
+import { PuppetRegistry } from "./tables/PuppetRegistry.sol";
+import { PUPPET_TABLE_ID } from "./constants.sol";
+import { Puppet } from "./Puppet.sol";
+
+contract PuppetMaster {
+  error PuppetMaster_NoPuppet(address systemAddress, ResourceId systemId);
+
+  function puppet() internal view returns (Puppet) {
```

Comment:
does this mean there's only a 1:1 mapping between a system and a puppet?

if I want multiple tokens (e.g. ERC721) on my world, how would I go about that?

Reply:
> does this mean there's only a 1:1 mapping between a system and a puppet

For now yes. Agree 1:N would be nicer but makes it much more complex, so I'd think about that later (either as a change to the puppet module, or with a new module)

> if I want multiple tokens (e.g. ERC721) on my world, how would I go about that?

You'll have to register different systems (see #1844). Thought a lot about this too, but I feel like this approach is the simplest (one namespace per token, with a separate system and tables in that namespace). Otherwise everything gets more complicated (if it's just one system you'd need to pass a param to the system, which means the interface doesn't match anymore with the puppet, access management becomes more complex, etc)